File: redcap_v15.7.6/API/examples/r/import_instrument_event_maps.R
#!/usr/bin/env Rscript

source('config.R')
library(RCurl)
library(jsonlite)

record <- list(
	arm=list(
		number=1,
		event=list(
			list(
				unique_event_name='event_1_arm_1',
				form=list('instr_1', 'instr_2')
			),
			list(
				unique_event_name='event_2_arm_1',
				form=list('instr_1')
			)
		)
	)
)

data <- toJSON(record, auto_unbox=TRUE)

result <- postForm(
    api_url,
    token=api_token,
    content='formEventMapping',
    format='json',
    data=data
)
print(result)

File: redcap_v15.7.6/API/examples/ruby/import_instrument_event_maps.rb
#!/usr/bin/env ruby

require 'json'
require 'curl'
require './settings.rb'
include Settings

record = {
  :arm => {
    :number => 1,
    :event => [
      {
        :unique_event_name => 'event_1_arm_1',
        :form => ['instr_1', 'instr_2',]
      },
      {
        :unique_event_name => 'event_2_arm_1',
        :form => ['instr_1',]
      },
    ]
  }
}

data = [record].to_json

fields = {
  :token   => Settings::API_TOKEN,
  :content => 'formEventMapping',
  :format  => 'json',
  :data    => data,
}

ch = Curl::Easy.http_post(
  Settings::API_URL,
  fields.collect{|k, v| Curl::PostField.content(k.to_s, v)}
)
puts ch.body_str
